topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java

```diff
@@ -581,6 +594,15 @@ private PartitionedByError<ReconcilableTopic, CurrentState> findDifferentRf(Part
      * @param reconcilableTopics Reconcilable topic.
      */
     private void warnTooLargeMinIsr(List<ReconcilableTopic> reconcilableTopics) {
+        var isCustomSaslConfigSet = config.saslCustomConfigJson() != null && !config.saslCustomConfigJson().isBlank();
+        if (isCustomSaslConfigSet) {
+            LOGGER.warnOp("Custom SASL config is being used. Some managed Kafka services like AWS MSK " +
+                  "don't allow reading cluster config. So the " + MIN_INSYNC_REPLICAS +
+                  " property is not being tested.");
+
+            return;
+        }
+
```

Comment:
This can be simplified like this:

```sh
if (config.saslCustomConfigJson() != null && !config.saslCustomConfigJson().isBlank()) {
    // some managed Kafka services don't allow reading cluster config
    return;
}
```

The problem with this is that some managed Kafka services may allow reading cluster config. So using an additional boolean env var could cover both use cases.

Reply:
I was focusing on the authentication and that is working now. However, now I am running into another issue when I crate a topic.

`errorMessage='Amazon MSK Serverless can't alter topic configuration segment.bytes. Only the following topic configurations are alterable: [compression.type, max.message.bytes, message.timestamp.difference.max.ms, message.timestamp.type, retention.bytes, retention.ms]. For all topic configurations, MSK Serverless uses defaults that follow best practices.`

I am thinking we add a config property to define the list of alterable properties, i.e. `STRIMZI_ALTERABLE_TOPIC_PROPERTIES`. Then in `BatchingTopicController.buildAlterConfigOps()` we can skip or remove any non-alterable values.  What do you think?

# Log below:
```
2024-03-27 02:34:19 DEBUG NetworkClient:934 - [AdminClient clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4] Received DESCRIBE_CONFIGS response from node 106 for request with header RequestHeader(apiKey=DESCRIBE_CONFIGS, apiVersion=4, clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4, correlationId=9, headerVersion=2): DescribeConfigsResponseData(throttleTimeMs=0, results=[DescribeConfigsResult(errorCode=0, errorMessage='', resourceType=2, resourceName='my-topic', configs=[DescribeConfigsResourceResult(name='compression.type', value='producer', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=2, documentation=null), DescribeConfigsResourceResult(name='leader.replication.throttled.replicas', value='', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=7, documentation=null), DescribeConfigsResourceResult(name='message.downconversion.enable', value='true', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=1, documentation=null), DescribeConfigsResourceResult(name='min.insync.replicas', value='2', readOnly=false, isDefault=false, configSource=4, isSensitive=false, synonyms=[], configType=3, documentation=null), DescribeConfigsResourceResult(name='segment.jitter.ms', value='0', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='partition.size.max.bytes', value='268435456000', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='cleanup.policy', value='delete', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=7, documentation=null), DescribeConfigsResourceResult(name='flush.ms', value='9223372036854775807', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='follower.replication.throttled.replicas', value='', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=7, documentation=null), DescribeConfigsResourceResult(name='segment.bytes', value='134217728', readOnly=false, isDefault=false, configSource=1, isSensitive=false, synonyms=[], configType=3, documentation=null), DescribeConfigsResourceResult(name='retention.ms', value='86400000', readOnly=false, isDefault=false, configSource=1, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='flush.messages', value='9223372036854775807', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='remote.log.msk.disable.policy', value='None', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=2, documentation=null), DescribeConfigsResourceResult(name='message.format.version', value='2.8-IV2', readOnly=false, isDefault=false, configSource=4, isSensitive=false, synonyms=[], configType=2, documentation=null), DescribeConfigsResourceResult(name='max.compaction.lag.ms', value='9223372036854775807', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='file.delete.delay.ms', value='60000', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='max.message.bytes', value='1048588', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=3, documentation=null), DescribeConfigsResourceResult(name='min.compaction.lag.ms', value='0', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='message.timestamp.type', value='CreateTime', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=2, documentation=null), DescribeConfigsResourceResult(name='preallocate', value='false', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=1, documentation=null), DescribeConfigsResourceResult(name='min.cleanable.dirty.ratio', value='0.5', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=6, documentation=null), DescribeConfigsResourceResult(name='index.interval.bytes', value='4096', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=3, documentation=null), DescribeConfigsResourceResult(name='unclean.leader.election.enable', value='false', readOnly=false, isDefault=false, configSource=4, isSensitive=false, synonyms=[], configType=1, documentation=null), DescribeConfigsResourceResult(name='retention.bytes', value='268435456000', readOnly=false, isDefault=false, configSource=4, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='delete.retention.ms', value='86400000', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='segment.ms', value='86400000', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='message.timestamp.difference.max.ms', value='9223372036854775807', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=5, documentation=null), DescribeConfigsResourceResult(name='segment.index.bytes', value='10485760', readOnly=false, isDefault=false, configSource=5, isSensitive=false, synonyms=[], configType=3, documentation=null)])])

2024-03-27 02:34:19 DEBUG BatchingTopicController:1137 - Reconciliation #2(upsert) KafkaTopic(platform-base/my-topic): Config changes [AlterConfigOp{opType=DELETE, configEntry=ConfigEntry(name=segment.bytes, value=null, source=UNKNOWN, isSensitive=false, isReadOnly=false, synonyms=[], type=UNKNOWN, documentation=null)}]

2024-03-27 02:34:19 DEBUG BatchingTopicController:1103 - Reconciliation #2(upsert) KafkaTopic(platform-base/my-topic): No partition change

2024-03-27 02:34:19 DEBUG BatchingTopicController:832 - Admin.incrementalAlterConfigs({ConfigResource(type=TOPIC, name='my-topic')=[AlterConfigOp{opType=DELETE, configEntry=ConfigEntry(name=segment.bytes, value=null, source=UNKNOWN, isSensitive=false, isReadOnly=false, synonyms=[], type=UNKNOWN, documentation=null)}]})

2024-03-27 02:34:19 DEBUG KafkaAdminClient:1534 - [AdminClient clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4] Queueing Call(callName=incrementalAlterConfigs, deadlineMs=1711506919377, tries=0, nextAllowedTryMs=0) with a timeout 30000 ms from now.

2024-03-27 02:34:19 DEBUG KafkaAdminClient:1277 - [AdminClient clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4] Sending IncrementalAlterConfigsRequestData(resources=[AlterConfigsResource(resourceType=2, resourceName='my-topic', configs=[AlterableConfig(name='segment.bytes', configOperation=1, value=null)])], validateOnly=false) to b106-nwid6xov.c1.kafka-serverless.us-east-1.amazonaws.com:9098 (id: 106 rack: null). correlationId=10, timeoutMs=30000

2024-03-27 02:34:19 DEBUG NetworkClient:545 - [AdminClient clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4] Sending INCREMENTAL_ALTER_CONFIGS request with header RequestHeader(apiKey=INCREMENTAL_ALTER_CONFIGS, apiVersion=1, clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4, correlationId=10, headerVersion=2) and timeout 30000 to node 106: IncrementalAlterConfigsRequestData(resources=[AlterConfigsResource(resourceType=2, resourceName='my-topic', configs=[AlterableConfig(name='segment.bytes', configOperation=1, value=null)])], validateOnly=false)

2024-03-27 02:34:19 DEBUG NetworkClient:934 - [AdminClient clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4] Received INCREMENTAL_ALTER_CONFIGS response from node 106 for request with header RequestHeader(apiKey=INCREMENTAL_ALTER_CONFIGS, apiVersion=1, clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4, correlationId=10, headerVersion=2): IncrementalAlterConfigsResponseData(throttleTimeMs=0, responses=[AlterConfigsResourceResponse(errorCode=40, errorMessage='Amazon MSK Serverless can't alter topic configuration segment.bytes. Only the following topic configurations are alterable: [compression.type, max.message.bytes, message.timestamp.difference.max.ms, message.timestamp.type, retention.bytes, retention.ms]. For all topic configurations, MSK Serverless uses defaults that follow best practices.', resourceType=2, resourceName='my-topic')])

2024-03-27 02:34:19 DEBUG KafkaAdminClient:1534 - [AdminClient clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4] Queueing Call(callName=listPartitionReassignments, deadlineMs=1711506919393, tries=0, nextAllowedTryMs=0) with a timeout 30000 ms from now.

2024-03-27 02:34:19 DEBUG KafkaAdminClient:1277 - [AdminClient clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4] Sending ListPartitionReassignmentsRequestData(timeoutMs=30000, topics=[ListPartitionReassignmentsTopics(name='my-topic', partitionIndexes=[0, 1, 2])]) to b63-nwid6xov.c1.kafka-serverless.us-east-1.amazonaws.com:9098 (id: 63 rack: null). correlationId=11, timeoutMs=30000

2024-03-27 02:34:19 DEBUG NetworkClient:545 - [AdminClient clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4] Sending LIST_PARTITION_REASSIGNMENTS request with header RequestHeader(apiKey=LIST_PARTITION_REASSIGNMENTS, apiVersion=0, clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4, correlationId=11, headerVersion=2) and timeout 30000 to node 63: ListPartitionReassignmentsRequestData(timeoutMs=30000, topics=[ListPartitionReassignmentsTopics(name='my-topic', partitionIndexes=[0, 1, 2])])

2024-03-27 02:34:19 DEBUG NetworkClient:934 - [AdminClient clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4] Received LIST_PARTITION_REASSIGNMENTS response from node 63 for request with header RequestHeader(apiKey=LIST_PARTITION_REASSIGNMENTS, apiVersion=0, clientId=strimzi-topic-operator-68ded8a6-a0f6-43dd-ae3a-5e8c2a352ec4, correlationId=11, headerVersion=2): ListPartitionReassignmentsResponseData(throttleTimeMs=0, errorCode=0, errorMessage='', topics=[])

2024-03-27 02:34:19 WARN  BatchingTopicController:1163 - Reconciliation #2(upsert) KafkaTopic(platform-base/my-topic): Reconciliation failed: org.apache.kafka.common.errors.InvalidConfigurationException: Amazon MSK Serverless can't alter topic configuration segment.bytes. Only the following topic configurations are alterable: [compression.type, max.message.bytes, message.timestamp.difference.max.ms, message.timestamp.type, retention.bytes, retention.ms]. For all topic configurations, MSK Serverless uses defaults that follow best practices.
```

Reply:
Yeah, maybe call it `STRIMZI_ALTERABLE_CONFIGURATIONS`. 
@tombentley are you good with this solution?

Reply:
@fvaleri @tombentley I was thinking about this. How about we add custom so it dovetails with the custom_sasl property? So people are less likely to throw it in willy nilly. `STRIMZI_CUSTOM_ALTERABLE_CONFIGURATIONS`

Reply:
TBH, I don't see this configuration strictly related to `SASL_CUSTOM_CONFIG_PROPERTIES`. I bet there are use cases not using SASL authentication, but that would still require `STRIMZI_ALTERABLE_TOPIC_CONFIG` (let's add back topic and use "config" like in the other env vars). Please make sure to also add documentation about this.